src/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -318,11 +250,10 @@ private void FlushMessageQueue()
         // We will lose some information if we don't do this.
         internal void WaitUtilEOF()
         {
-            if (_eofEvent != null)
+            if (_readToBufferTask != null)
             {
-                _eofEvent.WaitOne();
-                _eofEvent.Dispose();
-                _eofEvent = null;
+                _readToBufferTask.Wait();
```

Comment:
I'd suggest changing this to be .GetAwaiter().GetResult() instead of .Wait().  The only difference here is around what exception propagates out if the task is faulted, and that way if we do decide to let the exception from the user callback propagate out here, it'll be the exact same exception object rather than being wrapped in an aggregate.